packages/chat-client/src/controllers/engine.ts

```diff
@@ -790,30 +788,40 @@ export class ChatEngine extends IChatEngine {
 
       //TODO: Delete after 3 settled invites
       await this.client.chatSentInvites.update(topic, {
-        status: "accepted",
+        status: "approved",
       });
 
-      this.client.emit("chat_joined", {
+      this.client.emit("chat_invite_accepted", {
         id: payload.id,
-        topic: chatThreadTopic,
+        topic: topic,
+        params: {
+          invite: this.client.chatSentInvites.get(topic),
+          topic,
+        },
       });
     } else if (isJsonRpcError(payload)) {
       this.client.logger.error(payload.error);
       if (payload.error.message === getSdkError("USER_REJECTED").message) {
-        console.log("rejected invite... deleting", payload);
-        this.onRejectedChatInvite({ topic });
+        this.onRejectedChatInvite({ topic, id: payload.id });
       }
     }
   };
 
   protected onRejectedChatInvite: IChatEngine["onRejectedChatInvite"] = async ({
+    id,
     topic,
   }) => {
     await this.client.chatSentInvites.update(topic, {
       status: "rejected",
     });
 
-    console.log("reject > chatThreadsPending.delete:", topic);
+    this.client.emit("chat_invite_rejected", {
+      id: id,
+      topic: topic,
+      params: {
+        invite: this.client.chatSentInvites.get(topic),
```

Comment:
I saw this was in the spec but I'm curious: how come there's no explicit topic param in the rejection scenario? Or maybe the other way: why is it needed in the accepted one?

Reply:
It's needed as it replaced `chat_joined`, giving the possibility for wallets (like our own sample wallets) switch to a chat once it's been accepted. In the rejection case, there's no actual thread topic, so it's not supplied

Reply:
Thanks 🙏